src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,217 @@
+#include "RadialGeoMultiscaleMapping.hpp"
+#include <Eigen/src/Core/Matrix.h>
+#include <algorithm>
+#include <numeric>
+#include "mesh/Mesh.hpp"
+
+namespace precice::mapping {
+
+RadialGeoMultiscaleMapping::RadialGeoMultiscaleMapping(
+    Constraint     constraint,
+    int            dimensions,
+    MultiscaleType type,
+    MultiscaleAxis axis)
+    : Mapping(constraint, dimensions, false, Mapping::InitialGuessRequirement::None),
+      _type(type), _axis(axis)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void RadialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  size_t const inSize  = input()->vertices().size();
+  size_t const outSize = output()->vertices().size();
+
+  int effectiveCoordinate = static_cast<std::underlying_type_t<MultiscaleType>>(_axis);
+  PRECICE_ASSERT(effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::X) ||
+                     effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::Y) ||
+                     effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::Z),
+                 "Unknown multiscale axis type.")
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+    if (_type == MultiscaleType::SPREAD) {
+      // Determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+      Eigen::VectorXd axisMidpoints(inSize);
+      auto &          inputVerticesRef = input()->vertices();
+      // Order the vertices of the 1D input mesh, to correctly compute the midpoints of the segments between neighboring vertices
+      std::vector<size_t> ordered_vertex_indices(input()->vertices().size());
+      std::iota(ordered_vertex_indices.begin(), ordered_vertex_indices.end(), 0);
+      std::sort(ordered_vertex_indices.begin(), ordered_vertex_indices.end(),
```

Comment:
Do we need to preserve order of elements with the same effectiveCoordinate? If so, then we need `std::stable_sort` here.

Reply:
I don't think we can have such an issue on a 1D mesh, but I don't see any reason not to go for the safer `std::stable_sort`, which would also be a good example for further development steps.

Since this is happening only once in the beginning, and the 1D mesh should anyway be very coarse compared to anything else, performance is not an issue here.